scripts/install_protostar_rust_bindings.sh

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env bash
+
+set -e
+
+# clean up
+if [ "$1" == "--cleanup" ]; then
+  poetry env info -p | xargs rm -rf
+  if [[ $(uname -m) == 'arm64' ]]; then
+    CFLAGS=-I/opt/homebrew/opt/gmp/include LDFLAGS=-L/opt/homebrew/opt/gmp/lib poetry install
+  else
```

Comment:
I think this should rather check for "Darwin", as this is mac specific (and not arm specific)
```suggestion
  if [[ $(uname -s) == 'Darwin' ]]; then
    CFLAGS=-I/opt/homebrew/opt/gmp/include LDFLAGS=-L/opt/homebrew/opt/gmp/lib poetry install
  else
```

Reply:
I remember we talked about this when I originally came up with it and we decided to leave it as is for the time being. I was thinking we could address this separately but you made me think about this and this is actually a bug. The problem here is when we do `poetry run poe install_all_bindings --cleanup` it will execute both scripts with the `--cleanup` flag so it will discard the first installation.

As for your comment itself, I think you're wrong, this is actually arm-specific as it is a way to go for the m1/m2 Mac (you can see it in the protostar readme).

Reply:
> I remember we talked about this when I originally came up with it and we decided to leave it as is for the time being. I was thinking we could address this separately but you made me think about this and this is actually a bug. The problem here is when we do poetry run poe install_all_bindings --cleanup it will execute both scripts with the --cleanup flag so it will discard the first installation.

I suppose, but there is no such invokation on CI, right? This is only relevant to local dev env, and I suppose dev should know about it and if one wants to, be able to do the cleanup for either version; thus I don't see it as a bug, but as a feature ( ͡° ͜ʖ ͡°). 

> As for your comment itself, I think you're wrong, this is actually arm-specific as it is a way to go for the m1/m2 Mac (you can see it in the protostar readme).

- arm64 is not mac specific, and you **normally** won't have homebrew installation on linux 
- if I remember correctly I had this issue once on mac x86
- protostar readme is not the best source for this kind of information, as it is written by us
- cairo-lang docs mention macs in general, not m1/m2 (https://www.cairo-lang.org/docs/quickstart.html)

Reply:
> This is only relevant to local dev env, and I suppose dev should know about it

I think we should prevent such situations as much as we can ;)

As for the second part, I think you're partially right but not entirely ;) In fact, this is both arm and Mac specific, please note that the condition determines if we want to use the `CFLAGS`, the rest is the same (arm/m1 specific), But you're right that we use brew-related stuff (max specific).

Anyway, I removed the `--cleanup` option in [remove cleanup](https://github.com/software-mansion/protostar/pull/1881/commits/0a921e548daf44c60afbf4d381ddb14783033686). The reason is that we've been skeptical about this since the very beginning and also I don't really remember using it (probably no one does) so it may bring more confusion and problems than gain.

Reply:
> As for the second part, I think you're partially right but not entirely ;)

ok fair enough, if you're sure it is that way let's add checks for darwin && arm64 :+1:

>Anyway, I removed the --cleanup option in [remove cleanup](https://github.com/software-mansion/protostar/pull/1881/commits/0a921e548daf44c60afbf4d381ddb14783033686).

I'm ok with that :+1: